plugin/src/main/cpp/register_types.cpp

```diff
@@ -158,32 +264,66 @@ void initialize_plugin_module(ModuleInitializationLevel p_level) {
 			add_plugin_project_settings();
 
-			Engine::get_singleton()->register_singleton("OpenXRFbPassthroughExtensionWrapper", OpenXRFbPassthroughExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbRenderModelExtensionWrapper", OpenXRFbRenderModelExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSceneCaptureExtensionWrapper", OpenXRFbSceneCaptureExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSpatialEntityExtensionWrapper", OpenXRFbSpatialEntityExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSpatialEntityStorageExtensionWrapper", OpenXRFbSpatialEntityQueryExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSpatialEntityQueryExtensionWrapper", OpenXRFbSpatialEntityQueryExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSpatialEntityContainerExtensionWrapper", OpenXRFbSpatialEntityContainerExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbSceneExtensionWrapper", OpenXRFbSceneExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbHandTrackingAimExtensionWrapper", OpenXRFbHandTrackingAimExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbHandTrackingCapsulesExtensionWrapper", OpenXRFbHandTrackingCapsulesExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRFbCompositionLayerSettingsExtensionWrapper", OpenXRFbCompositionLayerSettingsExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRHtcFacialTrackingExtensionWrapper", OpenXRHtcFacialTrackingExtensionWrapper::get_singleton());
-			Engine::get_singleton()->register_singleton("OpenXRHtcPassthroughExtensionWrapper", OpenXRHtcPassthroughExtensionWrapper::get_singleton());
-
-			ClassDB::register_class<OpenXRFbRenderModel>();
-			ClassDB::register_class<OpenXRFbHandTrackingMesh>();
-			ClassDB::register_class<OpenXRFbSceneManager>();
-			ClassDB::register_class<OpenXRFbSpatialAnchorManager>();
-			ClassDB::register_class<OpenXRFbSpatialEntity>();
-			ClassDB::register_class<OpenXRFbSpatialEntityBatch>();
-			ClassDB::register_class<OpenXRFbSpatialEntityQuery>();
-			ClassDB::register_class<OpenXRFbSpatialEntityUser>();
-			ClassDB::register_class<OpenXRFbPassthroughGeometry>();
-			ClassDB::register_class<OpenXRMetaPassthroughColorLut>();
-
-			OpenXRFbHandTrackingAimExtensionWrapper::get_singleton()->add_project_setting();
+			if (global_bool_get("xr/openxr/extensions/fb_passthrough")) {
+				Engine::get_singleton()->register_singleton("OpenXRFbPassthroughExtensionWrapper", OpenXRFbPassthroughExtensionWrapper::get_singleton());
+				ClassDB::register_class<OpenXRFbPassthroughGeometry>();
+				ClassDB::register_class<OpenXRMetaPassthroughColorLut>();
+			}
```

Comment:
For the node types, I don't think we should make registering them conditional on the project setting, otherwise you have to restart the Godot editor for the node types to become available to use. So, I think this one, and all the following shouldn't be checking project settings

There may be an argument for always registering the classes for the extension wrappers also just not calling `register_extension_wrapper()`, so that the editor is aware that the classes exist, and what their methods are, because a couple of those singletons have public methods.

Reply:
yeah I need to dig into this, check all the safety checks are there